### ISSUE TITLE
Azure monitor with diagnosticsettings does not output ContainerAppConsoleLogs for crashing containers

### ISSUE THREAD

**Issue by csteeg** [microsoft/azure-container-apps#699]
### This issue is a: (mark with an x)
- [x ] bug report
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
If i configure the logging on the App Environment with destination: 'azure-monitor' and configure it to have 'allLogs' and 'audit' (or just allLogs) forwarded to a Log Analytics Workspace, ContainerAppConsoleLogs is created, but no entries appear from containers that don't startup due to errors on startup. That makes it hard to figure out why the container isn't starting.
If I configure the Environment to directly log to the workspace instead ('log-analytics'), the ContainerAppsConsoleLogs_CL contains the output of the failing container's console log

### Steps to reproduce
1. Configure azure monitor with a Log analytics workspace endpoint on an Apps Environment
2. Add a containerapp that has an image that crashes on startup and logs the error to the console
3. Wait for the error to appear in ContainerAppConsoleLogs 

**Expected behavior** [What you expected to happen.]
Log the console output to ContainerAppConsoleLogs just as the target: log-analytics does for ContainerAppConsoleLogs_CL

**Actual behavior** [What actually happened.]
No logs for crashing containers in ContainerAppConsoleLogs

**Comment by BigMorty** [comment]
@csteeg, if you set your environment to use Log Analytics directly rather than Azure Monitor --> Log Analytics, do you see the console log output you expect?

**Comment by csteeg** [comment]
@BigMorty indeed, then it gets written as expected in the ContainerAppConsoleLogs_CL

**Comment by BigMorty** [comment]
Thanks for confirming. @sanchitmehta, can you take a look at this please?
